server/src/main/java/com/objectcomputing/checkins/services/reviews/ReviewPeriodServicesImpl.java

```diff
@@ -346,4 +358,109 @@ private void createReviewRequest(ReviewPeriod period,
+                    }
+                }
+            }
+        }
+    }
+
+    void sendSelfReviewEmail(UUID reviewPeriodId, SelfReviewDate date) {
+        Optional<ReviewPeriod> reviewPeriod =
+            reviewPeriodRepository.findById(reviewPeriodId);
+        if (reviewPeriod.isEmpty()) {
+            LOG.error("Unable to find review period: " + reviewPeriodId.toString());
+            return;
+        }
+
+        // Determine which subject we need to use.
+        String subject = "";
+        switch(date) {
+            case SelfReviewDate.LAUNCH:
+                subject = reviewPeriod.get().getName() + " has launched!";
+                break;
+            case SelfReviewDate.THREE_DAYS:
+                subject = reviewPeriod.get().getName() +
+                          " closes in three days!";
+                break;
+            default:
+            case SelfReviewDate.ONE_DAY:
+                subject = reviewPeriod.get().getName() + " closes in one day!";
+                break;
+        }
+
+        try {
+            // Read in the email template.
+            String template = IOUtils.readText(
+                                  new BufferedReader(
+                                      selfReviewReminderTemplate.asReader()));
+
+            // Get the set of self-reviewer email addresses.
+            Set<MemberProfile> recipients = new HashSet<>();
+            String templateId = null;
+            List<FeedbackRequest> requests =
+                feedbackRequestRepository.findByValues(null, null, null, null,
+                                                       reviewPeriodId.toString(),
+                                                       templateId);
+            for (FeedbackRequest request : requests) {
+                if (request.getRecipientId().equals(request.getRequesteeId())) {
+                    Optional<MemberProfile> requesteeProfile =
+                        memberProfileRepository.findById(
+                            request.getRequesteeId());
+                    if (!requesteeProfile.isEmpty()) {
+                        recipients.add(requesteeProfile.get());
+                    }
+                }
+            }
+
+            for(MemberProfile recipient : recipients) {
+                // Customize the email content using the template.
+                String content = String.format(template,
+                                     webAddress,
+                                     reviewPeriodId.toString(),
+                                     recipient.getFirstName(),
+                                     dateAsString(reviewPeriod.get()
+                                                    .getSelfReviewCloseDate()),
+                                     webAddress);
+
+                // Send out the email to the individual.
+                emailSender.sendEmail(null, null, subject, content,
+                                      recipient.getWorkEmail());
+            }
```

Comment:
I think we may run into an issue here eventually as well. There are limits to the number of emails we can send. So, if we have an email template that does not require customization for a specific recipient, we should send it to all of them simultaneously (the sender batches them to 49 recipients per email).

Reply:
Oops...my bad. This one does require customization per recipient.

Reply:
@mkimberlin If we take out the by-name-greeting, we could send them out in bulk.

Reply:
@mkimberlin No.  The link just has the review period id in it.